### PR TITLE
CU-8685u1cq8_Council-add-DoWhile-runner

### PR DESCRIPTION
**Pull request by Naccomy**

### CODE REVIEW

**Comment by aflament** [comment]
Task linked: [CU-8685u1cq8 [Council] add DoWhile runner](https://app.clickup.com/t/8685u1cq8)

**Review by gkoch78 — CHANGES_REQUESTED** [review]
Let great overall. 
I'd like to see on more test about the predicate throwing :)

File: council/runners/do_while_runner.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+from council.contexts import ChainContext, ChatMessage
+from council.runners import RunnerBase, RunnerPredicate, RunnerExecutor, RunnerPredicateError
+
+
+class DoWhile(RunnerBase):
+    """
+    Runner that executes while the given predicate returns `True`
+    """
```
Comment (Naccomy) [inline council/runners/do_while_runner.py:8]:
```suggestion
    """
    Runner that executes one time and then while the given predicates returns `True`.
    """
```

File: council/runners/do_while_runner.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    def __init__(self, predicate: RunnerPredicate, runner: RunnerBase):
+        """
+        Args:
+            predicate: a predicate function
+            runner: a runner to be executed while the predicate returns `True`
```
Comment (Naccomy) [inline council/runners/do_while_runner.py:14]:
```suggestion
            runner: a runner to be executed one time and then while the predicate returns `True`
```

File: council/runners/do_while_runner.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
+from council.contexts import ChainContext, ChatMessage
+from council.runners import RunnerBase, RunnerPredicate, RunnerExecutor, RunnerPredicateError
+
+
+class DoWhile(RunnerBase):
+    """
+    Runner that executes while the given predicate returns `True`
```
Comment (gkoch78) [inline council/runners/do_while_runner.py:7]:
```suggestion
    Runner that executes an inner Runner while the given predicate returns `True`. The predicate is executed at the end of loop. As such, the inner runner executes at least once.
```

**Review by gkoch78 — CHANGES_REQUESTED** [review]
please add to the class diagram here: https://council--94.org.readthedocs.build/en/94/reference/runners/do_while.html

**Review by gkoch78 — APPROVED** [review]
🚀

**Pull request merged**